app/src/main/java/com/infomaniak/mail/utils/MessageBodyUtils.kt

```diff
@@ -55,61 +56,78 @@ object MessageBodyUtils {
 
         val htmlDocumentWithQuote = Jsoup.parse(initialBody.value)
         val htmlDocumentWithoutQuote = Jsoup.parse(initialBody.value)
-        val htmlQuotes = mutableListOf<Element>()
 
-        handleBlockQuote(htmlDocumentWithoutQuote, htmlQuotes)
-        val currentQuoteDescriptor = handleQuoteDescriptors(htmlDocumentWithoutQuote, htmlQuotes).ifEmpty { blockquote }
+        val blockquoteElement = findAndRemoveLastParentBlockquote(htmlDocumentWithoutQuote)
+        val currentQuoteDescriptor = findFirstKnownParentQuoteDescriptor(htmlDocumentWithoutQuote).ifEmpty {
+            if (blockquoteElement == null) "" else blockquote
+        }
 
-        val (body, quote) = splitBodyAndQuote(htmlQuotes, htmlDocumentWithQuote, currentQuoteDescriptor)
+        val (body, quote) = splitBodyAndQuote(blockquoteElement, htmlDocumentWithQuote, currentQuoteDescriptor)
```

Comment:
Shouldn't both of `splitBodyAndQuote` quote related arguments be next to each other rather than one being at the first position and one being at the third position ?

Reply:
The order does not matter, because there is not necessarily a link on these parameters of how they are managed. 
But no worries about changing it.